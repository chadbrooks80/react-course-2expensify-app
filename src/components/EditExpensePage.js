import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {editExpense, removeExpense} from '../actions/expenses';

export class EditExpensePage extends React.Component {
  
  constructor(props) {
    super(props);
  }

  onSubmit = expense => {
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  }
  
  onClick = () => {
    this.props.removeExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  }

  render() {
    return (
      <div>
        <ExpenseForm
          expense={this.props.expense}
          onSubmit={this.onSubmit}
        />
        <button onClick={this.onClick}>Remove</button>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  }
};

const mapDispatchToProps = (dispatch, props) => ({
  removeExpense: expenseId => dispatch(removeExpense(expenseId)),
  editExpense: (expenseId, expense) => dispatch(editExpense(expenseId, expense))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
