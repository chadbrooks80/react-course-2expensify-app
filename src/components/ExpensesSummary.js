import React from 'react';
import { connect } from 'react-redux'
import expensesTotal from '../selectors/expensesTotal';
import filterExpenses from '../selectors/expenses';
import numeral from 'numeral';

export const ExpensesSummary = (props) =>  {
    
    const displayResults = props.count !== 0;
    const expenseName = props.count > 1 ? 'expenses' : 'expense';

    return (
        <div>
            {displayResults &&
                <h1>{props.count} {expenseName} Totalling {props.total}</h1>
            }
        </div>
    );
};

const mapStateToProps = (state) => {
    const filteredExpenses = filterExpenses(state.expenses, state.filters)
    return {
        count: filteredExpenses.length,
        total: numeral(expensesTotal(filteredExpenses) / 100).format('$0,0.00')
    }
}

export default connect(mapStateToProps)(ExpensesSummary);