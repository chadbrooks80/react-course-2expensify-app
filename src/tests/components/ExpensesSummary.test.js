import React from 'react';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import { shallow } from 'enzyme';

test('snapshot should not display any results with zero expenses', () => {
    const wrapper = shallow(<ExpensesSummary count={0}  total={0} />);
    expect(wrapper).toMatchSnapshot();
});

test('snapshot should display with "expense" when expense count === 1', () => {
    const wrapper = shallow(<ExpensesSummary count={1}  total={1000} />);
    expect(wrapper).toMatchSnapshot();
});

test('snapshot should display with "expenses" when expense count > 1', () => {
    const wrapper = shallow(<ExpensesSummary count={2}  total={1500} />);
    expect(wrapper).toMatchSnapshot();
});