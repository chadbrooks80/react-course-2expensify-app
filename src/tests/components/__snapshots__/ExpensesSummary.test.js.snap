// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should display with "expense" when expense count === 1 1`] = `
<div>
  <h1>
    1
     
    expense
     Totalling 
    1000
  </h1>
</div>
`;

exports[`snapshot should display with "expenses" when expense count > 1 1`] = `
<div>
  <h1>
    2
     
    expenses
     Totalling 
    1500
  </h1>
</div>
`;

exports[`snapshot should not display any results with zero expenses 1`] = `<div />`;
